pkg/sriov/sriov.go

```diff
@@ -238,9 +244,14 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 	}
 
 	// 1. Set vlan
-	if conf.Vlan != 0 {
-		if err = s.nLink.LinkSetVfVlan(pfLink, conf.VFID, conf.Vlan); err != nil {
-			return fmt.Errorf("failed to set vf %d vlan: %v", conf.VFID, err)
+	if conf.Vlan != 0 || conf.VlanQoS != 0 {
```

Comment:
The `VlanQoS` is depends on 'vlan` i.e. it's not possible to set vlan qos if vlan is not provided. I think you'll need to move the check for VlanQos inside  
```
if conf.Vlan != 0 {
    // 1. Set vlan
    // 2. Set vlan qos if it's non-zero
}
```
block.

Also, we need to check if vlanQos is given in the conf without vlan then plugin should return an error with appropriate message. You can do this check inside LoadConf. It's a good idea to document the usage in the configuration section too.

Reply:
Actually, that was my initial implementation, however later I found that `ip link` accepts non-zero vlan qos while vlan id is set to zero (e.g. `ip link set eth0 vf 0 vlan 0 qos 5`). Though on the i40e VFs the packets weren't tagged with the PCP information - I wasn't sure what to do and decided to keep things simpler.

I'll update this block to your proposed logic, thanks.